src/p2/interpreter.rs

```diff
@@ -741,6 +745,102 @@ impl ParsedDocument {
         self.processing_imports = false;
     }
 
+    fn ignore_comments(&mut self) -> ftd::p1::Result<()> {
+        let mut processed_sections: Vec<ftd::p1::Section> = Vec::new();
+        for section in self.sections.iter() {
+            // Ignore entire section if it is commented out
+            if section.is_commented {
+                continue;
+            }
+
+            let mut filtered_section = ftd::p1::Section {
+                name: section.name.clone(),
+                caption: section.caption.clone(),
+                header: Default::default(),
+                body: None,
+                sub_sections: Default::default(),
+                is_commented: section.is_commented,
+                line_number: section.line_number,
+            };
```

Comment:
Use section as `mut` only. `.iter_mut()` would do it. No need to create this variable